### PR TITLE
Split cdds-rust into ddsc and libddsc-sys crates

### PR DESCRIPTION
**Pull request by esteve**
This PR splits the `cdds-rust` package into the `ddsc` and `libddsc-sys` crates according to https://doc.rust-lang.org/cargo/reference/build-scripts.html#-sys-packages

Eventually we could bundle CycloneDDS in `libddsc-sys`, but for now using the system version is ok.

### CODE REVIEW

**Comment by esteve** [comment]
@kydos this is now ready for review

**Review by kydos — COMMENTED** [review]
Using **ddsc** as the module name is inconsistent with other DDS API. Ideally, give a Rust binding to a DDS C-API, the resulting Rust API would be independent of the DDS vendor. Thus I am inclined to keep consistency with API in other languages and keep dds.

File: examples/pub.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use dds::Participant;
+use ddsc::Participant;
```
Comment (kydos) [inline examples/pub.rs]: This should remain **dds** as per the top level comment above.

**Comment by esteve** [comment]
@kydos I named the native package `libddsc-sys` because the system library is called `libddsc.so`, and so I named the high-level package `ddsc` to follow the Rust naming convention of using the native module name sans `lib` and `-sys`. Should I use `dds` for the module, but keep the package names as they are in this PR (i.e. `ddsc` and `libddsc-sys`)?

**Comment by esteve** [comment]
@kydos would it make sense to have a third module called `dds` that used/abstracted `ddsc` (and any other DDS implementation)? Is the API in the native `libddsc` library Cyclone-specific or is it a DDS standard?

**Comment by kydos** [comment]
> @kydos I named the native package `libddsc-sys` because the system library is called `libddsc.so`, and so I named the high-level package `ddsc` to follow the Rust naming convention of using the native module name sans `lib` and `-sys`. Should I use `dds` for the module, but keep the package names as they are in this PR (i.e. `ddsc` and `libddsc-sys`)?

Yes, let's use `dds` for the module name. For what concerns adding another layer, I am not a great fun as it makes layer on-top of layer to deal with implementations that do not comply with the DDS standard.

**Comment by esteve** [comment]
@kydos done in ea1a761

**Pull request merged**